File: apps/fogc/src/pump/many.cpp
#include "../type.hpp"

Defs with();
size_t with_count();

int many(Address address, Payload payload, Options options) {
    int count = 0;
    size_t total = with_count();
    Defs defs = with();
    for (size_t i = 0; i < total; i++) {
        if (defs[i].fit(address, payload, options)) {
            defs[i].fun(address, payload, options);
            count++;
        }
    }
    return count;
}

File: apps/fogr/src/pump/many.rs
use crate::defs::with::with;
use crate::r#type::{Address, Options, Payload};

pub fn many(address: Address, payload: Payload, options: Options) -> i32 {
    let mut count = 0;
    for def in with() {
        if (def.fit)(address, payload, options) {
            (def.fun)(address, payload, options);
            count += 1;
        }
    }
    count
}
